core/templates/pages/exploration-editor-page/modal-templates/save-version-mismatch-modal.component.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for version mismatch modal.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { LoggerService } from 'services/contextual/logger.service';
+import { ExplorationDataService } from 'pages/exploration-editor-page/services/exploration-data.service';
+import { LostChange, LostChangeObjectFactory } from 'domain/exploration/LostChangeObjectFactory';
+
+@Component({
+  selector: 'oppia-save-version-mismatch-modal',
+  templateUrl: './save-version-mismatch-modal.component.html'
+})
+export class SaveVersionMismatchModalComponent implements OnInit {
+  MSECS_TO_REFRESH: number = 20;
+  hasLostChanges: boolean;
+  @Input() lostChanges: LostChange[];
+
+  constructor(
+    private windowRef: WindowRef,
+    private loggerService: LoggerService,
+    private explorationDataService: ExplorationDataService,
+    private lostChangeObjectFactory: LostChangeObjectFactory
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.hasLostChanges = (this.lostChanges && this.lostChanges.length > 0);
+    if (this.hasLostChanges) {
+      // TODO(sll): This should also include changes to exploration
+      // properties (such as the exploration title, category, etc.).
+      this.lostChanges = this.lostChanges.map(
```

Comment:
Ditto, this should be done at the caller

Reply:
Resolved as per offline discussion.